vms/evm/predicate/README.md

```diff
@@ -0,0 +1,141 @@
+
+
+# Predicate
+
+This package contains the predicate data structure and its encoding and helper functions to unpack/pack the data structure.
+
+## Encoding
+
+A byte slice of size N is encoded as:
+
+1. Slice of N bytes
+2. Delimiter byte `0xff`
+3. Appended 0s to the nearest multiple of 32 bytes
+
+
+## Results
+
+`predicate_results.go` defines how to encode `PredicateResults` within the block header's `Extra` data field.
```

Comment:
nit: I would just drop the go-file name reference here. It's so easy to get this out-of-sync with the file names.